openproblems/data/immune_cells.py

```diff
@@ -0,0 +1,56 @@
+from . import utils
+
+import os
+import scanpy as sc
+import scprep
+import tempfile
+
+URL = "https://ndownloader.figshare.com/files/25717328"
+
+
+@utils.loader
+def load_immune(test=False):
+    """Download immune human data from figshare."""
+    if test:
+        # load full data first, cached if available
+        adata = load_immune(test=False)
+
+        # Subsample pancreas data
+        adata = adata[:, :500].copy()
+        utils.filter_genes_cells(adata)
+
+        # We are using the final annotation column from the original file
+        # for the cell type annotation, the batch column contains the name
+        # of the study the data originates from
+
+        keep_final_annotations = adata.obs["final_annotation"].dtype.categories[[0, 3]]
+        keep_batchs = adata.obs["batch"].dtype.categories[[0, -3, -2]]
```

Comment:
please document which batches you are removing here for test data and how many are left